packages/nx/src/command-line/dep-graph.ts

```diff
@@ -465,28 +474,27 @@ function debounce(fn: (...args) => void, time: number) {
   };
 }
 
-function createFileWatcher(root: string, changeHandler: () => Promise<void>) {
+async function createFileWatcher(
+  root: string,
+  changeHandler: () => Promise<void>
+) {
   const ignoredGlobs = getIgnoredGlobs(root);
-  const layout = workspaceLayout();
 
-  const watcher = watch(
-    [
-      joinPathFragments(layout.appsDir, '**'),
-      joinPathFragments(layout.libsDir, '**'),
-    ],
-    {
-      cwd: root,
-      ignoreInitial: true,
-    }
-  );
-  watcher.on(
-    'all',
-    debounce(async (event: string, path: string) => {
-      if (ignoredGlobs.ignores(path)) return;
-      await changeHandler();
+  const unregisterFileWatcher = await daemonClient.registerFileWatcher(
+    { watchProjects: 'all', includeGlobalWorkspaceFiles: true },
+    debounce(async (error, { changedFiles }) => {
+      if (error === 'closed') {
+        output.error({ title: `Watch error: Daemon closed the connection` });
+        process.exit(1);
+      } else if (error) {
+        output.error({ title: `Watch error: ${error?.message ?? 'Unknown'}` });
+      } else if (changedFiles.some(({ path }) => !ignoredGlobs.ignores(path))) {
```

Comment:
Not sure if the ignored check is needed here or if the daemon handles this internally. I have to check again.

Reply:
Dameon watcher already handles the ignored files. So this can be removed.